src/components/editor-page/editor-pane/linter/single-line-regex-linter.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { Linter } from './linter'
+import type { EditorView } from '@codemirror/view'
+import type { Diagnostic } from '@codemirror/lint'
+import { t } from 'i18next'
+
+export class SingleLineRegexLinter implements Linter {
+  constructor(
+    private regex: RegExp,
+    private message: string,
+    private replace: (match: string) => string,
+    private actionLabel?: string
+  ) {}
+
+  lint(view: EditorView): Diagnostic[] {
```

Comment:
This method is very long. You could make it more readable by extracting stuff into separate functions like the creation of the diagnostic.